splice_machine/src/main/java/com/splicemachine/derby/utils/Vacuum.java

```diff
@@ -199,6 +199,9 @@ public void vacuumDatabase(long oldestActiveTransaction) throws SQLException{
 
             List<DataDelete> deletes = new ArrayList<>();
             for (long removed : toDelete) {
+                boolean restoreMode = SIDriver.driver().lifecycleManager().isRestoreMode();
+                if (restoreMode)
+                    break;
```

Comment:
Should we break out here or return directly? It is possible that when restoreMode becomes true, the _deletes_ list is non-empty, in this case, do we want "dropped.delete(deletes)" to be executed or not?

Reply:
Good point. We may want to terminate vacuum immediately during a database restore.